PE 546 SFI Page Redesign

vuphuctho opened the pull request:
- [x] Redo SFI design
- [x]  UI responsiveness (mobile display for non-responsive components like Tab)
- [x] Display correct data from backend

shuesken commented:
Great @vuphuctho! Noticing some superficial issues looking at [`caf`:](http://localhost:3000/sustainable-eco-banks/caf)

---

Serving shouldn't display `[]` when empty (although I don't know what it _should_ display)
<img width="387" alt="image" src="https://github.com/bank-green/nuxt-bank-green/assets/32738398/7605dc6a-be1f-4720-a9b6-d4def8773d5b">

----

It should say "Mortgage **or** Loans"
<img width="733" alt="image" src="https://github.com/bank-green/nuxt-bank-green/assets/32738398/59115a28-1d03-45e0-842b-f35b9ca5c9da">

---

Spacing between bullet points is too tight compared to the mockup:
<img width="1119" alt="image" src="https://github.com/bank-green/nuxt-bank-green/assets/32738398/63eb6d39-c808-4192-9965-4ef4ff51cdbb">

--- 

The stock image aren't added yet either, but I believe that's addressed that in the Linear comment from albert: https://linear.app/bankgreen/issue/PE-546#comment-a992f2d4

---

I don't see a mockup for the mobile page, but to me, this spacing between the sections looks too big:
<img width="451" alt="image" src="https://github.com/bank-green/nuxt-bank-green/assets/32738398/46c87676-f87d-4277-a09d-1cf718b9ea38">

---

I think it'd be useful to have consistency between kebab-case and camelCase in the component prop passing both in `EcBankHeader` and `EcoBankDetail`

vuphuctho commented:
@shuesken 

> - Serving shouldn't display [] when empty (although I don't know what it should display)
> - It should say "Mortgages or Loans"
> - Spacing between bullet points is too tight compared to the mockup:

I have fixed above issues in recent commit

> I think it'd be useful to have consistency between kebab-case and camelCase in the component prop passing both in `EcoBankHeader` and `EcoBankDetail`

Since we use camelCase in our code, I have modified all kebab-case to camelCase

vuphuctho commented:
> Spacing between bullet points is too tight compared to the mockup:
Added the styling fix. Though if we can control it via Prismic, I think it would be better.

vuphuctho commented:
@shuesken @RogerTangos For page without prismic content, for now we display `fromTheWebsite` like so:

page URL: http://localhost:3000/sustainable-eco-banks/amalgamated_bank
Current design
![Screenshot 2023-05-12 at 14-47-07 Find Ethical   Sustainable Banks In Your Area - Bank Green](https://github.com/bank-green/nuxt-bank-green/assets/3623903/81b8d6eb-b9aa-4097-b4e5-7677be6db0fe)

New design 
![Screenshot 2023-05-12 at 14-46-44 Amalgamated Bank Review and Service Offering - Bank Green](https://github.com/bank-green/nuxt-bank-green/assets/3623903/f570828a-b6c1-45f7-b2cb-90e2f787860a)

shuesken commented:
@vuphuctho Lovely! I think the "Serving: []" thing is still there though. Please also remove the "Tell us if we got something wrong", it seems odd to have there. (This is just my design sense talking tho)

Otherwise looks good to me! Would love to @RogerTangos 's check on the design too, but I think we're good to go here otherwise.

Out of curiosity, why would you prefer having the styling in Prismic rather than in the frontend? Addding the `space-y-6` the way you did seems to me like exactly the right way of separating content and website design there.

vuphuctho commented:
@shuesken Sorry, my misunderstaning about 'Serving'. I have fixed it in last commit

About styling of Prismic content, right now we limit these fields to be bullet points, so adding `space-y-6` solves our styling requirement. But in the future, if we need to enable more content types in these section (paragraphs, multi-leveled bullet points) then refining the styling would be a hassle. If the content creator on Prismic does not need to contact our dev side if they want to introduce any new type of content but can fully control from their side, it would be better for them.

However, just my two cent. Since our team is small, I don't see it as an important requirement.

RogerTangos commented:
I'm still seeing the issue with "serving" here. Zak and I talked about this after refining the ticket and I made the mistake of not re-editing it. **Instead of using "serving" from django, can we please use whatever text is currently in prismic?** That's probably easier to implement than using the data from django anyways, it separates concerns, and gives content editors more flexibility 

<img width="730" alt="Screenshot 2023-05-04 at 10 19 54" src="https://github.com/bank-green/nuxt-bank-green/assets/1609475/06ab4fc0-2f36-464d-be85-1b849053a7d6">

RogerTangos commented:
Another issue: I believe that the accredidation images (with the exception of fossil free certified) should now be served via the prismic SFIDefaults types. i.e. `institution_credentials-bankers for net zero`

RogerTangos commented:
I fixed the "serving" issue. The accredidations still look a bit wonky to me though. For example, Unity Trust Bank is both a part of GABV and Bankers for Net Zero, but the image doesn't appear for me. I'd prefer to get the `ecoBankHeader` displaying accredidations directly from the SFIDefaults page if possible, but they really should work at launch.

RogerTangos commented on components/eco-bank/EcoBankHeader.vue:
```diff
@@ -23,22 +29,27 @@
                     <span v-else-if="rating === 'ok'">This bank is ok</span>
                     <span v-else-if="rating === 'great'">This bank is great</span>
                 </div>
+                <div class="flex justify-start items-center gap-6 mt-4" v-if="hasInstitutionCredentials">
```
should use prismic if possible. Alt should also be pulled from prismic

vuphuctho commented:
@RogerTangos I reverted [your last change]([https://github.com/bank-green/nuxt-bank-green/pull/48/commits/e17779114a3978f1da6dde3b4e0383f6cc43be2a) in order to display `serving` with `PrismicText` instead.

vuphuctho commented:
@RogerTangos the accreditation images are now used Prismic resource instead of local ones.

RogerTangos commented:
Approved. DO NOT MERGE YET. Merging will not take place until other in-sprint tickets have been completed.
